packages/vue-components/src/Navbar.vue

```diff
@@ -71,12 +75,114 @@
+          return false;
+        }
+        for (let i = 0; i < hParts.length - 1; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      isChild(url, href) {
+        const hParts = this.splitUrl(href);
+        const uParts = this.splitUrl(url);
+        if (uParts.length <= hParts.length) {
+          return false;
+        }
+        for (let i = 0; i < hParts.length; i++) {
+          if (hParts[i] !== uParts[i]) {
+            return false;
+          }
+        }
+        return true;
+      },
+      isExact(url, href) {
+        return this.normalizeUrl(url) === this.normalizeUrl(href);
+      },
+      highlightLink(url) {
+        const defHlMode = this.defaultHighlightOn;
+        const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
+        // attempt an exact match first
+        for (const li of navLis) {
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const hlMode = a.getAttribute('data-highlight') || defHlMode;
+            if (hlMode === 'none') {
+              continue;
+            }
+            // terminate early on an exact match
+            if (this.isExact(url, a.href)) {
+              li.classList.add('current');
+              return;
+            }
+          }
+        }
+        // fallback to user preference, otherwise
+        for (const li of navLis) {
+          const navLinks = Array.from(li.querySelectorAll('a.nav-link'));
+          const dropdownLinks = Array.from(li.querySelectorAll('a.dropdown-item'));
+          const allNavLinks = navLinks.concat(dropdownLinks).filter(a => a.href);
+          for (const a of allNavLinks) {
+            const hlMode = a.getAttribute('data-highlight') || defHlMode;
+            if (hlMode === 'none') {
+              continue;
+            }
+            if (hlMode === 'exact') {
+              if (this.isExact(url, a.href)) {
+                li.classList.add('current');
+                return;
+              }
+            } else if (hlMode === 'sibling-or-child') {
+              if (this.isSibling(url, a.href) || this.isChild(url, a.href)) {
+                li.classList.add('current');
+                return;
+              }
+            } else if (hlMode === 'sibling') {
+              if (this.isSibling(url, a.href)) {
+                li.classList.add('current');
+                return;
+              }
+            } else if (hlMode === 'child') {
+              if (this.isChild(url, a.href)) {
+                li.classList.add('current');
+                return;
+              }
+            } else {
+              // invalid option, ignore
+              return;
```

Comment:
```suggestion
              console.log(something);
```

so we can continue checking the rest of the links with valid options (if any)